gen/test-html.hs

```diff
@@ -21,23 +22,68 @@ data Presentation = Presentation
     { title :: String
     , author :: String
     , tags :: [Tag]
-    , slides :: Maybe String
-    , audio :: Maybe String
-    , player :: Maybe String
+    , slides :: Maybe FilePath
+    -- ^ Relative path to slides.
+    , audio :: Maybe FilePath
+    -- ^ Relative path to audio recording of this presentation.
+    , player :: Maybe FilePath
+    -- ^ Relative path to web player able to handle audio recording of this
+    -- presentation.
     } deriving Show
 
+instance Default Presentation where
+    def = Presentation
+        { title = "Unknown title"
+        , author = "Unknown author"
+        , tags = []
+        , slides = Nothing
+        , audio = Nothing
+        , player = Nothing
+        }
+
 data Meetup = Meetup
     { indexM :: Integer
+    -- ^ Meetups form a total order in time and is mapped in to linear ordering
+    -- of 'Integer' numbers starting from 0.
     , presentations :: [Presentation]
+    -- ^ List of presentations presented during meetup.
     , time :: Maybe ZonedTime
+    -- ^ Time when the meetup occurred
     , participants :: Maybe Integer
+    -- ^ Number of participats, for future, or meetups currently being held,
+    -- this is set, obviously, to 'Nothing'.
     } deriving Show
 
+instance Default Meetup where
+    def = Meetup
+        { indexM = -1
+        , presentations = []
+        , time = Nothing
+        , participants = Nothing
+        }
+
+-- | Assign index to a 'Meetup'. It expects them to be in reverse order, i.e.
+-- newest/future first and oldest as last.
+indexMeetups :: [Meetup] -> [Meetup]
```

Comment:
Ok, makes sense. How about switching to smart constructor for a future meetup? Something like:

````Haskell
futureMeetup :: Integer -> Meetup
futureMeetup idx = Meetup
        { indexM = idx
        , presentations = []
        , time = Nothing
        , participants = Nothing
        }
````

That would also mean getting rid of "instance Default Meetup".

I would also change `indexMeetups` in to `checkMeetupIndexes` and instead of numbering them it would just check them and throw error if the indexes aren't correctly ordered. Can we agree on that?

From my experience people will most likely screw up small things. In this case using a particular number. Big things, like adding new meetup, are less likely to be done incorrectly and we have an advantage of code review. This discussion is a proof of that. :-)